src/AddressRegistry.Api.BackOffice/Handlers/CreateStreetNameSnapshotSqsHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace AddressRegistry.Api.BackOffice.Handlers
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Abstractions;
+    using Abstractions.SqsRequests;
+    using Be.Vlaanderen.Basisregisters.Sqs;
+    using Be.Vlaanderen.Basisregisters.Sqs.Handlers;
+    using TicketingService.Abstractions;
+
+    public sealed class CreateStreetNameSnapshotSqsHandler : SqsHandler<CreateStreetNameSnapshotSqsRequest>
+    {
+        public const string Action = "CreateStreetNameSnapshot";
+
+        private readonly BackOfficeContext _backOfficeContext;
+
+        public CreateStreetNameSnapshotSqsHandler(
+            ISqsQueue sqsQueue,
+            ITicketing ticketing,
+            ITicketingUrl ticketingUrl,
+            BackOfficeContext backOfficeContext)
+            : base(sqsQueue, ticketing, ticketingUrl)
+        {
+            _backOfficeContext = backOfficeContext;
+        }
+
+        protected override string? WithAggregateId(CreateStreetNameSnapshotSqsRequest request)
+        {
+            return _backOfficeContext
+                .AddressPersistentIdStreetNamePersistentIds
+                .SingleOrDefault(x => x.StreetNamePersistentLocalId == request.Request.StreetNamePersistentLocalId)
+                ?.StreetNamePersistentLocalId.ToString();
```

Comment:
De aggregate ID is onderdeel van het request. Dit enkel gedaan om te verifiëren of de straatnaam bestaat. Indien null, dan wordt de except AggregateIdNotFound gegooid. Deze wordt opgevangen in API.